### PR TITLE
HBASE-24222 remove FSUtils.checkAccess and replace with FileSystem.access in HBCK

### PR DESCRIPTION
**Pull request by nyl3532016**

### CODE REVIEW

File: hbase-hbck2/src/main/java/org/apache/hbase/hbck1/HBaseFsck.java [hunk 1]
```diff
@@ -2450,7 +2450,7 @@ private void checkAndFixTableStates() throws IOException {
     }
   }
 
-  private void preCheckPermission() throws IOException, AccessDeniedException {
+  private void preCheckPermission() throws IOException, AccessControlException {
```
Comment (jojochuang) [inline hbase-hbck2/src/main/java/org/apache/hbase/hbck1/HBaseFsck.java:2453]: In printUsageAndExit() which invokes preCheckPermission(), should it also update the following catch clause to catch AccessControlException instead?
`
    } catch (AccessDeniedException ace) {
`

Comment (nyl3532016) [reply hbase-hbck2/src/main/java/org/apache/hbase/hbck1/HBaseFsck.java:2453]: printUsageAndExit do not invoke preCheckPermission ? I can not find  the invoke

Comment (liuml07) [reply hbase-hbck2/src/main/java/org/apache/hbase/hbck1/HBaseFsck.java:2453]: The only place I found was `HBaseFsck.java`, which deals with this the same as generic IOE (simply `exit`)

**Comment by nyl3532016** [comment]
@liuml07 @jojochuang please hep merge this one

**Comment by liuml07** [comment]
> @liuml07 @jojochuang please help merge this one

I don’t have commit access. We can ping @Apache9 to have a look.

**Comment by jatsakthi** [comment]
+1. Thanks for the contribution @nyl3532016 and the review @liuml07 & @jojochuang . Let me merge this one.

**Pull request merged**